### ISSUE TITLE
Bug in autoloader for PHP library for Drive REST API

### ISSUE THREAD

**Issue by grant** [googleapis/google-api-php-client#1647]
> Note: This issue was moved from `b/23349670`, reported on 8/19/2015.

Reported by an external friend:

--snip--

Basically, in this API of googles, https://developers.google.com/api-client-library/php/start/installation it’s for the PHP library for Drive REST API - it states that you need PHP 5.2.1 or greater to use the autoloader.php, well this is WRONG.
 
If you look at the autoloader.php how it is coded is for PHP 5.3.0 or greater, thus in order to make the autoloader work for 5.2.1 (as it states) you must change the code in it being this way: http://php.net/manual/en/function.spl-autoload-register.php look at example #1. Or here is the corrected code:
 
```php
    function autoloader ($className) {
      $classPath = explode('_', $className);
      if ($classPath[0] != 'Google') {
        return;
      }
      // Drop 'Google', and maximum class file path depth in this project is 3.
      $classPath = array_slice($classPath, 1, 2);
 
      $filePath = dirname(__FILE__) . '/' . implode('/', $classPath) . '.php';
      if (file_exists($filePath)) {
        require_once($filePath);
      }
    }
    spl_autoload_register('autoloader');
 ```

--snip--

**Comment by jdpedrie** [comment]
I don't believe this is relevant anymore. We no longer include our own autoloader, autoloading is instead delegated to composer. Furthermore, PHP 5.3 and below are not supported by the client. This issue is related to the fact that anonymous functions were not available priot to PHP 5.3.

@dwsupplee anything I may be missing, or can this be closed?

**Comment by dwsupplee** [comment]
Yes, it looks like we can close this out.
